webapp/courses/blockparser.py

```diff
@@ -55,13 +56,58 @@ def le(self):
     "anchor": Tag("a", "[[", "]]", re.compile(r"\[\[(?P<address>.+?)([|](?P<link_text>.+?))?\]\]")),
     "kbd":    Tag("kbd", "`", "`", re.compile(r"`(?P<kbd>.+?)`")),
     "hint":   Tag("mark", "[!hint=hint_id!]", "[!hint!]", re.compile(r"\[\!hint\=(?P<hint_id>\w+)\!\](?P<hint_text>.+?)\[\!hint\!\]")),
-    "term":   Tag("span", '"""', '"""', re.compile(r"\"{3}(?P<term>.+?)\"{3}")),
+    "term":   Tag("span", '[?term=term_id?]', '[?term?]', re.compile(r"\[\?term\=(?P<term_id>\w+)\?\](?P<term_text>.+?)\[\?term\?\]")),
```

Comment:
Use ”!” instead of a ”?” here. So, [!term= ...]. This [!something=something!] ... [!something!] is going to be used for other miscellaneous stuff too.
Also – since the end user is not accessing the term by its id, something like `term_name` should be used here instead of `term_id`.